### DISCUSSION TITLE
Add "Any Terrain", "Ignore", and "Other" as options for terrain tile interactions

### DISCUSSION THREAD

**Discussion by Daedalus1400** [godotengine/godot-proposals#4896] [2D]
When setting up rules for terrain tiles in 4.0, the only options are "No tile" and "Terrain Set <x>".  There is no way to group possibilities for faster setup.

If one wanted to make a particular tile ignore whether there is an adjacent tile in a particular position then they would need to make an alternative tile for every possible combination of tiles at that location.  The number of alternatives needed would grow exponentially with the number of terrain sets present.  Even with only one terrain set, the number of alternatives is unnecessarily large.  The experience is similar for any grouping such as "any terrain tile" and "any tile from a different terrain set".

Adding "Any Terrain", "Ignore", and "Other" as options would streamline the creation of tile set terrain rules.

Implementation could look something like this:

<pre><code>ANY = -2;
IGNORE = -3;
OTHER = -4:
</code></pre>

<pre><code>...
switch (rule) {
    case ANY:
        return otherTile.terrainSet > -1;
    case IGNORE:
        return true;
    case OTHER:
        return (otherTile.terrainSet > -1 && otherTile.terrainSet != thisTile.terrainSet);
    default:
        return otherTile.terrainSet == rule;
}
...
</code></pre>

**Comment by Hexogrammer** [comment]
Being able to specify a specific collection of terrain sets would be even better, I think
